src/parser/metrics/Metric.ts

```diff
@@ -79,28 +79,22 @@ export abstract class SourceFile {
      */
     filePath: string;
 
-    /**
-     * File extension of the file.
-     */
-    fileExtension: string;
-
-    protected constructor(filePath: string, fileExtension: string) {
+    protected constructor(filePath: string) {
         this.filePath = filePath;
-        this.fileExtension = fileExtension;
     }
 }
 
 /**
  * Represents a file written in an unsupported language.
  */
 export class UnsupportedFile extends SourceFile {
-    constructor(filePath: string, fileExtension: string) {
-        super(filePath, fileExtension);
+    constructor(filePath: string) {
+        super(filePath);
     }
 }
 
 /**
- * Represents a supported file to be analyzed for metrics, including its path, file extension, language and parsed syntax tree.
+ * Represents a supported file to be analyzed for metrics, including its path, language and parsed syntax tree.
```

Comment:
Do you think it is really necessary to mention all the attributes in the comment? This is a coupling and you have to update the comment all the time.

Reply:
No, I can change that.

Reply:
Changed.